pkg/controller/perconaservermongodb/status.go

```diff
@@ -173,7 +173,11 @@ func (r *ReconcilePerconaServerMongoDB) updateStatus(cr *api.PerconaServerMongoD
 	case replsetsPaused == len(repls):
 		cr.Status.State = api.AppStatePaused
 	case !inProgress && replsetsReady == len(repls) && clusterState == api.AppStateReady && cr.Status.Host != "":
-		cr.Status.State = api.AppStateReady
+		if cr.Spec.Sharding.Enabled && cr.Status.Mongos.Status != api.AppStateReady {
+			cr.Status.State = cr.Status.Mongos.Status
+		} else {
```

Comment:
we don't need else if we keep old 176 line

Reply:
fixed: https://github.com/percona/percona-server-mongodb-operator/pull/696/commits/87f2f517688eac8dd529eebdd582b9512a71bada